### ISSUE TITLE
Screen readers does not announce state and position count on controls present at the left navigation in windows.

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#32188]
### Describe the bug

**User Impact:**
Visually Impaired users who uses screen readers will be impacted as Screen readers (Narrator) does not announce state and position count on controls present at the left navigation in windows.

**Pre-Requisites:**
Enable Narrator

**Actual Result:** 
Screen readers (Narrator) does not announce state and position count on controls present at the left navigation in windows.
Narrator announces: Link Docs

**Refer Attachment:**
1.Screen readers (Narrator) does not announce state and position count on controls present at the left navigation in windows.png
2.Screen readers (Narrator) does not announce state and position count on controls present at the left navigation in windows.mp4

<img width="969" height="546" alt="Image" src="https://github.com/user-attachments/assets/42afeffc-3814-4616-9ecb-d5e1dc13b62b" />
https://github.com/user-attachments/assets/29b4bb54-d88e-414d-8036-54dd44def19f

**Expected Result:**
Screen readers (Narrator) should announce state as selected/ not selected and position count as 1 of 18 on controls present at the left navigation in windows.

### Reproduction link

https://vivacommonstorybooktest.z13.web.core.windows.net/?path=/docs/app-bar--docs

### Reproduction steps

1. Open URL: https://vivacommonstorybooktest.z13.web.core.windows.net/?path=/docs/app-bar--docs in edge browser in windows.
2. 'App Bar - Docs' page will be open.
3. Press tab key to navigate to the 'Docs' control present at the left navigation.
4. Verify whether screen reader announces the state and position count on controls present at the left navigation in windows or not.

### System

```bash
OS: Windows 11 24H2 (Build - 26407.5000)
Browser: Edge Version 138.0.3351.121
Screen reader: Narrator
Storybook 7.6.20
```

**Comment by Sidnioulz** [comment]
Hi @msftedad!

Thanks for taking the time to write a report. We're aware of many accessibility issues in the sidebar, and I'll be rewriting it in the coming weeks. You can follow https://github.com/storybookjs/storybook/issues/31267 to track that work.

We'll likely use GitHub's TreeView, which has been user-tested for accessibility. You can find a few examples below if you'd like to see how it works:
* https://primer.style/react/storybook/iframe.html?globals=&args=&id=components-treeview-features--focus-management&viewMode=story
* https://primer.style/react/storybook/?path=/story/components-treeview-features--nested-trees
* also in use on the sub-issues section of https://github.com/storybookjs/storybook/issues/30587, where you can see how secondary actions are provided

If you'd like, I can let you know when a PR's open so you can test it yourself and provide feedback. It'll be based on Storybook 9.1, though. We no longer provide support for Storybook 7.

I'm gonna go ahead and close this issue as duplicated to limit noise in the issue list.
